### PR TITLE
A8s 2633 postgresql 16 update

### PR DESCRIPTION
**Pull request by mkfdoherty**
# Short Description

_Please provide a brief summary of your changes_

# Details

# Note

WARNING: Only users listed in the CODEOWNERS file can approve PRs!

# Checks

- [ ] Documentation has been adjusted
- [ ] Architectural decisions have been documented
- [ ] Changelog has been updated
- [ ] PR is approved by a code owner
- [ ] Manifests are updated
- [ ] Commit message adheres to our [guideline](https://anynines.atlassian.net/wiki/spaces/DS/pages/2423193626/Version+Control+Workflow)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ Changelog](https://keepachangelog.com/en/1.0.0/).
 ### Added
 
 * BackupPolicy feature: Create new backups based on a schedule. You can now configure a schedule for each Postgresql instance's backups and when the backups will be cleaned up.
+* postgresql-operator PostgreSQL 16 support
+* backup_agent PostgreSQL version 16 support and improved backup/restore reliability across versions.
```
Comment (lhaendler) [inline CHANGELOG.md]: I think this is an internal detail. Do we want to share it with our customers in a change log?

File: test/framework/parse.go [hunk 2]
```diff
@@ -37,9 +42,27 @@ func ParseEnv() (TestRunConfig, error) {
...
+	}
+	config.Version = version
 	return config, validateConfig(config)
 }
 
+func getIntEnv(key string, fallback int) (int, error) {
+	value := os.Getenv(key)
+	if value == "" {
+		return fallback, nil
+	}
```
Comment (lhaendler) [inline test/framework/parse.go]: Have you considered https://pkg.go.dev/os#LookupEnv?

File: CHANGELOG.md (hunk 1 above)
Comment (mkfdoherty) [reply CHANGELOG.md]: I take it you are referring to the backup_agent addition only? One could say that improving the robustness of backup/restore across versions is a meaningful change. But I don't feel too strongly and would agree with the removal of the CHANGELOG entry.

File: test/framework/parse.go (hunk 2 above)
Comment (mkfdoherty) [reply test/framework/parse.go]: I did consider it but it appeared to not add additional value for this implementation. WDYT?

File: CHANGELOG.md (hunk 1 above)
Comment (lhaendler) [inline CHANGELOG.md]:
```suggestion
* PostgreSQL 16 support
* Improved backup/restore reliability across versions.
```

Comment (lhaendler) [reply CHANGELOG.md]: I added an alternative suggestion, but this is non-blocking for me

File: docs/platform-operators/postgresql_version_migration.md
Comment (lhaendler) [inline docs/platform-operators/postgresql_version_migration.md:1]: Praise: You really went the extra mile with this one! Nice workaround :)

File: CHANGELOG.md
Comment (lhaendler) [inline CHANGELOG.md:1]: Suggestion: let's add a summary of the most notable changes in pg 16, or at least a link to their changelog

File: test/framework/parse.go (hunk 2 above)
Comment (lhaendler) [reply test/framework/parse.go]: Adding a safeguard against unset variables could help users catch configuration errors more easily. Defining a variable, but not setting a value commonly means that some templating has failed.

**Review by lhaendler — COMMENTED** [review]
Code looks good so far, I'm still holding of testing until everything is ready

File: test/framework/parse.go (hunk 2 above)
Comment (mkfdoherty) [reply test/framework/parse.go]: https://github.com/anynines/a8s-deployment/pull/342/commits/4b953418c2f6a41aa73a9854ca88c009818c9cee

**Comment by mkfdoherty** [comment]
> I added an alternative suggestion, but this is non-blocking for me

Where is the alternative suggestion?

File: CHANGELOG.md (hunk 1 above)
Comment (mkfdoherty) [reply CHANGELOG.md]: https://github.com/anynines/a8s-deployment/pull/342/commits/0f4be45ddd37c945828b726e51a20c8683aaa281

File: CHANGELOG.md
Comment (mkfdoherty) [reply CHANGELOG.md:1]: Good idea: https://github.com/anynines/a8s-deployment/pull/342/commits/22cd8201ebeb0da08472fc98bee007f2ddd5b0e0

**Comment by mkfdoherty** [comment]
I am rerunning the end-to-end (e2e) test with the fix to ensure it works correctly on our Tekton pipeline.

**Review by lhaendler — APPROVED** [review]
lgtm

**Pull request merged**